sources/web/datalab/polymer/components/datalab-settings/datalab-settings.ts

```diff
@@ -80,7 +80,10 @@ class SettingsElement extends Polymer.Element {
    */
   _themeChanged() {
     return SettingsManager.setUserSettingAsync('theme', this.theme)
-      .then(() => document.dispatchEvent(new Event('ThemeChanged')));
+      .then(() => {
+        const e = new CustomEvent('ThemeChanged', {'detail': this.theme})
+        document.dispatchEvent(e);
```

Comment:
Any reason not to add explicit set/get methods on SettingsManager and have a static 'x_changed' event on that? Generally seems better to avoid going through the global document for events.